pkg/frontend/transport/handler.go

```diff
@@ -329,20 +330,19 @@ func (f *Handler) reportQueryStats(
 		return
 	}
 	userID := tenant.JoinTenantIDs(tenantIDs)
-	var stats *querier_stats.Stats
-	var samplesProcessedFromCache uint64
+	var stats *querier_stats.QueryStats
+	var samplesProcessedCacheAdjusted uint64
 	if details != nil {
 		stats = details.QuerierStats
-		samplesProcessedFromCache = details.SamplesProcessedFromCache
+		samplesProcessedCacheAdjusted = details.SamplesProcessedCacheAdjusted
 	}
 	wallTime := stats.LoadWallTime()
 	numSeries := stats.LoadFetchedSeries()
 	numBytes := stats.LoadFetchedChunkBytes()
 	numChunks := stats.LoadFetchedChunks()
 	numIndexBytes := stats.LoadFetchedIndexBytes()
-	sharded := strconv.FormatBool(stats.GetShardedQueries() > 0)
+	sharded := strconv.FormatBool(stats.LoadShardedQueries() > 0)
```

Comment:
nice catch, but changes like these are sometimes easier to review in a separate PR, especially when it's bundled with a lot of changes like this PR

Reply:
Some tests were panicing without it, after I wrapped protogenerated stats into the "QueryStats".